src/main/kotlin/calculator/StringAddCalculator.kt

```diff
@@ -0,0 +1,48 @@
+package calculator
+
+object StringAddCalculator {
+
+    private val DELIMITER = listOf(",", ":")
+
+    fun sum(input: String?): Int {
+        if (input.isNullOrBlank()) {
+            return 0
+        }
+        val (addString, customDelimiter) = parseInputPattern(input)
+        val delimiter = getDelimitersRegex(customDelimiter)
+        val numbers = split(addString, delimiter)
+        if (hasNegative(numbers)) {
+            throw RuntimeException("음수는 입력할 수 없습니다.")
+        }
+        return sumAll(numbers)
+    }
+
+    private fun parseInputPattern(input: String): Pair<String, String?> {
+        var addPattern: String? = null
+        var customDelimiter: String? = null
+        val result = Regex("//(.)\n(.*)").find(input)
+        result?.let {
+            customDelimiter = it.groupValues[1]
+            addPattern = it.groupValues[2]
+        }
+
+        return Pair(addPattern ?: input, customDelimiter)
```

Comment:
```suggestion
        val result = Regex("//(.)\n(.*)").find(input) ?: return (input to null)
        return result.groupValues[1] to result.groupValues[2]
```
가변 변수 없이도 엘비스 연산자를 잘 활용하면 구현할 수 있을 것 같아요.

추가적으로 코멘트 남기면 Pair를 리턴하는 경우에 구현은 편하지만, 이를 사용하는 쪽에서 Pair의 양 값이 어떤 값임을 한 눈에 파악하기 어렵다는 단점이 있습니다. 특히 이 메서드가 재활용돼서 여러 곳에서 사용되기 시작하면 문제가 더 커지겠죠.
이런 경우에 DTO 성격의 data class를 하나 정의해서 사용하면 destructuring을 비롯한 기능들은 그대로 사용하면서 각 값에 이름을 붙일 수도 있습니다.